packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -0,0 +1,135 @@
+import {type Draft as WritableDraft} from '@reduxjs/toolkit';
+import {createReducer, FacetValueRequest} from '../../../../ssr.index';
+import {
+  toggleExcludeFacetValue,
+  toggleSelectFacetValue,
+  updateFacetIsFieldExpanded,
+  updateFacetNumberOfValues,
+} from '../../../facets/facet-set/facet-set-actions';
+import {convertFacetValueToRequest} from '../../../facets/facet-set/facet-set-slice';
+import {fetchProductListing} from '../../product-listing/product-listing-actions';
+import {
+  CommerceFacetSetState,
+  getCommerceFacetSetInitialState,
+} from './facet-set-state';
+import {CommerceFacetRequest} from './interfaces/request';
+import {AnyFacetResponse, FacetResponse} from './interfaces/response';
+
+export const commerceFacetSetReducer = createReducer(
+  getCommerceFacetSetInitialState(),
+
+  (builder) => {
+    builder
+      .addCase(fetchProductListing.fulfilled, (state, action) => {
+        const facets = action.payload.response.facets;
```

Comment:
I think we should also clear out facets that were not in the response. Otherwise, if we request PLP 1 and it has facet A, and then we subsequently request PLP 2 but it doesn't have facet A, facet A will still be in the state even though it shouldn't.

I believe this will become even more obviously required when we also have to support search in the same engine 😁 

We could either just wipe the facet state clean on fetchProductListing.fulfilled right before we do updateStateFromFacetResponse, or we could memo a list of what we've updated from the response and wipe the delta after (I think I prefer the latter option).

Reply:
Went with the second option! We keep a set of facets to delete, and remove from it as we iterate on the new set of facets. Once we've iterated over all facets, we remove all leftover items in the set of facets to delete.